cli/src/main/java/com/devonfw/tools/ide/tool/aws/Aws.java

```diff
@@ -0,0 +1,99 @@
+package com.devonfw.tools.ide.tool.aws;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.Set;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.environment.EnvironmentVariables;
+import com.devonfw.tools.ide.environment.EnvironmentVariablesType;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+
+/**
+ * {@link LocalToolCommandlet} for AWS CLI (aws).
+ *
+ * @see <a href="https://docs.aws.amazon.com/cli/">AWS CLI homepage</a>
+ */
+
+public class Aws extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Aws(IdeContext context) {
+
+    super(context, "aws", Set.of(TAG_CLOUD));
+  }
+
+  private void makeExecutable(Path file) {
+
+    // TODO this can be removed if issue #132 is fixed
+    Set<PosixFilePermission> permissions = null;
+    try {
+      permissions = Files.getPosixFilePermissions(file);
+      permissions.add(PosixFilePermission.GROUP_EXECUTE);
+      permissions.add(PosixFilePermission.OWNER_EXECUTE);
+      permissions.add(PosixFilePermission.OTHERS_EXECUTE);
+      Files.setPosixFilePermissions(file, permissions);
+    } catch (IOException e) {
+      throw new RuntimeException("Adding execution permission for Group, Owner and Others did not work for " + file, e);
+    }
+  }
+
+  @Override
+  protected void moveAndProcessExtraction(Path from, Path to) {
+
+    if (this.context.getSystemInfo().isLinux()) {
+      // make binary executable using java nio because unpacking didn't preserve the file permissions
+      // TODO this can be removed if issue #132 is fixed
+      Path awsInDistPath = from.resolve("dist").resolve("aws");
+      Path awsCompleterInDistPath = from.resolve("dist").resolve("aws_completer");
+      makeExecutable(awsInDistPath);
+      makeExecutable(awsCompleterInDistPath);
+
+      // running the install-script that aws shipped
+      ProcessContext pc = this.context.newProcess();
+      Path linuxInstallScript = from.resolve("install");
+      pc.executable(linuxInstallScript);
+      pc.addArgs("-i", from.toString(), "-b", from.toString());
+      pc.run();
```

Comment:
doesn't this require admin/root permissions? Maybe `sudo` required?

Reply:
most probably not as you are passing the `from` dir as installation target directory.
Shouldn't we install it in `to` dir and get rid of the sources or does it only work as a "in-situ" installation with a mix of source-code and compiled stuff?

Reply:
I have tested it on Ubuntu 22 and it worked there without ever asking for the sudo pw.

Reply:
> Here on linux the actual generic installation process is "completed" but only has an "installer" in software/aws and also created a .ide.tool.version. Therefore if the post installation fails, it will never recover from its inconsistent state.
Even worse the installer has already installed in the software repository and your post installation logic is somehow "in-situ" manipulating this result. Therefore when installing version X then installing Y and then installing version X again will result in symlinking the previous X version from the software repository where post install has already run on. So IMHO this will lead to a dead end and will never get stable and free of bugs.

You mentioned this in a previous comment. So I would install it to `from`, then create the symlinks, and after everything is done and has completed without error, move it to `to`.